src/sample.py

```diff
@@ -39,7 +40,7 @@ def message (self, buf, font, msg, data, _x_what_is_this):
 		return True
 debugger = Debugger()
 hb.buffer_set_message_func (buf, debugger.message, 1, 0)
-hb.buffer_add_utf8 (buf, text.encode('utf-8'), 0, -1)
+hb.buffer_add_codepoints (buf, codepoints, 0, -1)
```

Comment:
The [documentation](http://behdad.github.io/harfbuzz/harfbuzz-Buffers.html#hb-buffer-add-codepoints) mentions this, do you think it is not clear enough?